eslint.config.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
```

Comment:
![image](https://github.com/user-attachments/assets/6858cb5f-b1b1-469b-8185-e58a6a75dcb2)

кажется не везде соблюдаются правила линтера )

Reply:
по какой-то причине у меня не работает линтер
ошибка из eslint консоли: ConfigError: Config (unnamed): Key "env": This appears to be in eslintrc format rather than flat config format.
попробовал перевести на flat, тогда ругается при билде, и так же линтер не включается
подсветка тоже не работает
команда  npx eslint --config ./eslint.config.js ./src/index.ts выдает месседж: 0:0  warning  File ignored because no matching configuration was supplied
Можешь пожалуйста подсказать где неправильно сконфигурировал?

Reply:
попробовал чето такое собрать, вроде все норм)
можно попробовать от этого оттолкнуться

import tsPlugin from "@typescript-eslint/eslint-plugin";

export default [
  {
    files: ["*.ts", "*.tsx"],
    languageOptions: {
      parser: "@typescript-eslint/parser",
      parserOptions: {
        project: "./tsconfig.json",
        tsconfigRootDir: process.cwd(),
        ecmaVersion: "latest",
        sourceType: "module",
      },
    },
    plugins: {
      "@typescript-eslint": tsPlugin,
    },
    rules: {
      quotes: ["error", "double"],
      semi: ["error", "always"],
      indent: ["error", 2],
      "no-console": "error",
      ...tsPlugin.configs.recommended.rules,
      ...tsPlugin.configs["recommended-requiring-type-checking"].rules,
    },
  },
];

Reply:
только надо не забыть добавить `"type": "module"` в `package.json`